app/models/tenant_quota.rb

```diff
@@ -36,7 +36,9 @@ class TenantQuota < ApplicationRecord
 
   NAMES = QUOTA_BASE.keys.map(&:to_s)
 
-  validates :name, :inclusion => {:in => NAMES}
+  validates :name,
+            :inclusion  => {:in => NAMES},
+            :uniqueness => {:scope => :tenant_id, :message => "should be unique per tenant"}
```

Comment:
@imtayadeway you were right, no need to scope `:name` as we are already validating it. I added a test to validate uniqueness of quota per tenant.